docs/content/r&d/services/boot/uefi-boot-endpoints.md

```diff
@@ -0,0 +1,242 @@
+
+#### Sequence Diagram
+
+```mermaid
+sequenceDiagram
+    participant Client as Bare Metal Server
+    participant Boot as Boot Server
+    participant DB as Firestore
+    
+    Client->>Boot: GET /boot.ipxe?mac=52:54:00:12:34:56
+    Boot->>Boot: Validate MAC address format
+    Boot->>DB: Query machine by MAC
+    DB-->>Boot: Machine config (profile_id, kernel args)
+    Boot->>DB: Get profile by ID
+    DB-->>Boot: Profile config (image_id, kernel args)
+    Boot->>Boot: Generate iPXE script
+    Boot-->>Client: 200 OK (iPXE script)
+```
+
+**Query Parameters:**
+
+| Parameter | Type | Required | Description |
+|-----------|------|----------|-------------|
+| `mac` | string | Yes | MAC address of the requesting machine (format: `aa:bb:cc:dd:ee:ff`) |
+
+**Request Example:**
+
+```http
+GET /boot.ipxe?mac=52:54:00:12:34:56 HTTP/1.1
+Host: boot.internal
+```
+
+**Response Example (200 OK):**
+
+```text
+#!ipxe
+
+# Boot configuration for node-01 (52:54:00:12:34:56)
+# Profile: urn:boot:profile:ubuntu-22.04-server
+# Generated: 2025-11-19T06:00:00Z
+
+kernel /assets/urn:boot:image:ubuntu-2204/kernel console=tty0 console=ttyS0 ip=dhcp
+initrd /assets/urn:boot:image:ubuntu-2204/initrd
+boot
+```
+
+**Response Headers:**
+
+- `Content-Type: text/plain; charset=utf-8`
+- `Cache-Control: no-cache, no-store, must-revalidate`
+
+**Error Responses:**
+
+| Status Code | Description | Example |
+|-------------|-------------|---------|
+| 400 Bad Request | Missing or invalid MAC address | `{"error": {"code": "INVALID_MAC_ADDRESS", "message": "MAC address must be in format aa:bb:cc:dd:ee:ff"}}` |
+| 404 Not Found | No boot configuration found for MAC | `{"error": {"code": "MACHINE_NOT_CONFIGURED", "message": "No boot configuration found for MAC 52:54:00:12:34:56"}}` |
+| 500 Internal Server Error | Database or template error | `{"error": {"code": "INTERNAL_ERROR", "message": "Failed to generate boot script"}}` |
+
+**Boot Script Variables:**
+
+The iPXE script may include the following dynamic values:
+
+- Machine-specific kernel parameters
+- Asset download URLs (using URN format)
+- Network configuration parameters
+
+---
+
+## Kernel Image Endpoint
+
+### `GET /assets/{id}/kernel`
+
+Streams kernel images from Cloud Storage for the boot process.
+
+#### Sequence Diagram
+
+```mermaid
+sequenceDiagram
+    participant Client as Bare Metal Server
+    participant Boot as Boot Server
+    participant Storage as Cloud Storage
+    
+    Client->>Boot: GET /assets/urn:boot:image:ubuntu-2204/kernel
+    Boot->>Boot: Validate URN format
+    Boot->>Boot: Parse image ID from URN
+    Boot->>Storage: Stream kernel file
+    Storage-->>Boot: Kernel data stream
+    Boot-->>Client: 200 OK (kernel stream)
+```
+
+**Path Parameters:**
+
+| Parameter | Type | Required | Description |
+|-----------|------|----------|-------------|
+| `id` | string (URN) | Yes | Boot image identifier (URN format: `urn:boot:image:{name}`) |
```

Comment:
Instead of boot image identifier, this should be the id of the machine, as defined by the boot service.